typing/printtyp.ml

```diff
@@ -661,11 +669,12 @@ let penalty s =
 let rec path_size = function
     Pident id ->
       penalty (Ident.name id), -Ident.scope id
-  | Pdot (p, _) ->
+  | Pdot (p, _) | Pextra_ty (p, Pcstr_ty _) ->
       let (l, b) = path_size p in (1+l, b)
   | Papply (p1, p2) ->
       let (l, b) = path_size p1 in
       (l + fst (path_size p2), b)
+  | Pextra_ty (p, _) -> path_size p
```

Comment:
Here it would make sense to use `path_size p + 1` as in the `Pdot` case, but this change has the benefit of not changing the short-paths behavior.